### PR TITLE
Feature/171 Add tier 1 reattribution screen

### PR DESCRIPTION
**Pull request by mjtravers**
#171

### CODE REVIEW

File: front-end/src/app/reports/transactions/transaction-container/transaction-container.component.ts [hunk 1]
```diff
@@ -30,6 +32,15 @@ export class TransactionContainerComponent extends DestroyerComponent {
     if (isPulledForwardLoan(this.transaction)) {
       return 1;
     }
+    if (
+      this.activatedRoute.snapshot.queryParamMap.get('reattribution') ||
```
Comment (toddlees) [inline front-end/src/app/reports/transactions/transaction-container/transaction-container.component.ts]: checking the query params in the route should probably be the responsibility of the resolver.  the condition below (...isReattRedes...) would work for new and existing transactions if the resolver sets it up properly

File: front-end/src/app/reports/transactions/transaction-list/transaction-list-table-base.component.ts [hunk 2]
```diff
@@ -137,6 +137,15 @@ export abstract class TransactionListTableBaseComponent extends TableListBaseCom
         this.reportIsEditable,
       () => true
     ),
+    new TableAction(
+      'Reattribute',
+      this.createReattribution.bind(this),
+      (transaction: Transaction) =>
+        transaction.transactionType.scheduleId === ScheduleIds.A &&
```
Comment (toddlees) [inline front-end/src/app/reports/transactions/transaction-list/transaction-list-table-base.component.ts:141]: According to the ticket, it's only possible to reattribute tier 1's so we'll want to disable if they have a parent

**Review by toddlees — CHANGES_REQUESTED** [review]

File: front-end/src/app/shared/models/transaction-type.model.ts [hunk 3]
```diff
@@ -28,7 +28,8 @@ export abstract class TransactionType {
   contactConfig: { [contactKey: string]: { [formField: string]: string } } = STANDARD_SINGLE_CONTACT;
   abstract schema: JsonSchema; // FEC validation JSON schema
   abstract templateMap: TransactionTemplateMapType; // Mapping of values between the schedule (A,B,C...) and the common identifiers in the HTML templates
-  abstract getNewTransaction(): Transaction; // Factory method to create a new Transaction object with default property values for this transaction type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  abstract getNewTransaction(properties?: any): Transaction; // Factory method to create a new Transaction object with default property values for this transaction type
```
Comment (toddlees) [inline front-end/src/app/shared/models/transaction-type.model.ts]: is properties used anywhere?

File: front-end/src/app/reports/transactions/transaction-container/transaction-container.component.ts (hunk 1 above)
Comment (mjtravers) [reply front-end/src/app/reports/transactions/transaction-container/transaction-container.component.ts]: This line of code turned out to be OBE the following code and I was able to remove it.

File: front-end/src/app/reports/transactions/transaction-list/transaction-list-table-base.component.ts (hunk 2 above)
Comment (mjtravers) [reply front-end/src/app/reports/transactions/transaction-list/transaction-list-table-base.component.ts:141]: Good catch. I added a check for the parent_transaction_id

File: front-end/src/app/shared/models/reattribution-redesignation/reattribution-to.model.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      subTitle:
+        'The portion of an excessive contribution that has been attributed, in writing, to another contributor and signed by both contributors.',
+      accordionTitle: 'ENTER DATA',
+      accordionSubText: 'Reattribute a contribution or a portion of it to a different contributor.',
+      formTitle: 'Reattribution to',
+      contactTitle: 'Contact',
+      dateLabel: 'REATTRIBUTION DATE',
+      amountLabel: 'REATTRIBUTED AMOUNT',
+      footer:
+        'The information in this rattribution will automatically create a related receipt. Review the receipt, or continue without reviewing and "Save transactions."',
```
Comment (dheitzer) [inline front-end/src/app/shared/models/reattribution-redesignation/reattribution-to.model.ts]: Typo here.  Should be 'reattribution'

File: front-end/src/app/shared/resolvers/transaction.resolver.ts [hunk 5]
```diff
@@ -49,6 +59,35 @@ export class TransactionResolver {
...
+            ReattRedesTypes.REATTRIBUTION_FROM,
+            ReattRedesTypes.REDESIGNATION_TO,
+            ReattRedesTypes.REDESIGNATION_FROM,
+          ])
+        ) {
+          if (transaction.reatt_redes_id) {
+            return this.transactionService.getReattRedes(transaction.reatt_redes_id).pipe(
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              map((transactions: any | undefined) => {
+                const reattributed = transactions.reattributed;
```
Comment (dheitzer) [inline front-end/src/app/shared/resolvers/transaction.resolver.ts]: The resolver could fail here if undefined (404 from service for example) so prob need to add truthy check here to be graceful.

File: front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.spec.ts [hunk 6]
```diff
@@ -71,7 +71,7 @@ describe('TransactionTypeBaseComponent', () => {
     expect(component.transactionType?.title).toBe('Individual Receipt');
   });
 
-  it('should save on save event', fakeAsync(() => {
+  xit('should save on save event', fakeAsync(() => {
```
Comment (toddlees) [inline front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.spec.ts]: does this still fail?

**Review by Elaine-Krauss-TCG — APPROVED** [review]

**Review by toddlees — APPROVED** [review]

File: front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.spec.ts (hunk 6 above)
Comment (mjtravers) [reply front-end/src/app/shared/components/transaction-type-base/transaction-type-base.component.spec.ts]: This unit test has been fixed and is now back in the rotation.

File: front-end/src/app/shared/models/transaction-type.model.ts (hunk 3 above)
Comment (mjtravers) [reply front-end/src/app/shared/models/transaction-type.model.ts]: Ah, good catch. This is OBE code from a previous iteration of managing the reattribution overlay of properties. It has been removed.

**Pull request merged**